### PR TITLE
expression: migrate test-infra to testify for expression

### LINKED ISSUE #29133: migrate test-infra to testify for `expression/evaluator_test.go`

### LINKED ISSUE #29135: migrate test-infra to testify for `expression/expression_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: 

close #29133
close #29135

Problem Summary:

migrate test-infra to testify for expression/evaluator_test.go, expression/expression_test.go, expression/function_traits_test.go

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@tisonkun @AilinKid PTAL

File: expression/evaluator_test.go [hunk 1]
```diff
@@ -353,23 +316,31 @@ func (s *testEvaluatorSuite) TestBinopBitop(c *C) {
...
-		fc := funcs[t.op]
-		f, err := fc.getFunction(s.ctx, datumsToConstants(types.MakeDatums(t.lhs, t.rhs)))
-		c.Assert(err, IsNil)
+	ctx := mock.NewContext()
+	ctx.GetSessionVars().StmtCtx.TimeZone = time.Local
+	sc := ctx.GetSessionVars().StmtCtx
+	sc.TruncateAsWarning = true
+	err := ctx.GetSessionVars().SetSystemVar("max_allowed_packet", "67108864")
+	require.NoError(t, err)
+	ctx.GetSessionVars().PlanColumnID = 0
```
Comment (tisonkun) [inline expression/evaluator_test.go:325]: You may factor out these lines to a utility function `createEvaluatorSuite` or something.

Comment (hawkingrei) [reply expression/evaluator_test.go]: I find that ```createContext``` can init context instead of this code block.

Comment (tisonkun) [reply expression/evaluator_test.go]: Yep. I create it to supersede `SetUpTest` when migrating builtin tests 😉

**Review by tisonkun — APPROVED** [review]
LGTM.

**Comment by hawkingrei** [comment]
@tangenta @wjhuang2016 @AilinKid  PTAL

File: expression/evaluator_test.go [hunk 2]
```diff
@@ -489,38 +442,31 @@ func (s *testEvaluatorSuite) TestBinopNumeric(c *C) {
...
+		f, err := fc.getFunction(ctx, datumsToConstants(types.MakeDatums(tt.lhs, tt.rhs)))
+		require.NoError(t, err)
 		v, err := evalBuiltinFunc(f, chunk.Row{})
-		c.Assert(err, IsNil)
-		c.Assert(v.Kind(), Equals, types.KindNull)
+		require.NoError(t, err)
+		require.Equal(t, types.KindNull, v.Kind())
 	}
-
-	s.ctx.GetSessionVars().StmtCtx.InSelectStmt = oldInSelectStmt
```
Comment (wjhuang2016) [inline expression/evaluator_test.go:517]: Why not reset it?

Comment (hawkingrei) [reply expression/evaluator_test.go:517]: it doesn't reuse this ctx so that we don't have to reset it. for the next test, you create a new ctx to use.

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**